tests/models/utils.py

```diff
@@ -78,8 +117,64 @@ def check_logprobs_close(
     for prompt_idx, (outputs_0,
                      outputs_1) in enumerate(zip(outputs_0_lst,
                                                  outputs_1_lst)):
-        output_ids_0, output_str_0, logprobs_0 = outputs_0
-        output_ids_1, output_str_1, logprobs_1 = outputs_1
+        if len(outputs_0) == 3:
+            assert len(outputs_1) == 3
+            # Break out tokens, text & sample logprobs
+            # (prompt logprobs were not provided)
+            output_ids_0, output_str_0, logprobs_0 = outputs_0
+            output_ids_1, output_str_1, logprobs_1 = outputs_1
+        elif len(outputs_0) == 4:
+            assert len(outputs_1) == 4
```

Comment:
```suggestion
        assert len(output_0) == len(output_1)
        if len(outputs_0) == 3:
            # Break out tokens, text & sample logprobs
            # (prompt logprobs were not provided)
            output_ids_0, output_str_0, logprobs_0 = outputs_0
            output_ids_1, output_str_1, logprobs_1 = outputs_1
        elif len(outputs_0) == 4:
```

Reply:
@comaniac I reverted this suggested change because I remembered why I had placed the asserts in the way that I did - without the asserts, the `mypy` static analysis tool doesn't seem to realize that the if statements (`if len(outputs_0) == <3 or 4>`) constrain the `len()` of the `outputs_0` tuple, resulting in the following formatting failures

```
Success: no issues found in 127 source files
tests/models/utils.py:125: error: Too many values to unpack (3 expected, 4 provided)  [misc]
tests/models/utils.py:139: error: Need more than 3 values to unpack (4 expected)  [misc]
Found 2 errors in 1 file (checked 281 source files)
```

Switching to

```
        assert len(outputs_0) == len(outputs_1)
        if len(outputs_0) == 3:
            assert len(outputs_1) == 3
            # Break out tokens, text & sample logprobs
            # (prompt logprobs were not provided)
            output_ids_0, output_str_0, logprobs_0 = outputs_0
            output_ids_1, output_str_1, logprobs_1 = outputs_1
        elif len(outputs_0) == 4:
            assert len(outputs_1) == 4
            # Break out tokens, text, sample logprobs & prompt logprobs
            (
                output_ids_0,
                output_str_0,
                logprobs_0,
                prompt_logprobs_0,
            ) = outputs_0
            (
                output_ids_1,
                output_str_1,
                logprobs_1,
                prompt_logprobs_1,
            ) = outputs_1
```

seems to give `mypy` enough info to perform the static analysis (i.e. there are no formatting errors.)

Reply:
Oops ok. We have to listen to mypy in the first place lol